packages/cloud/src/queries/tenants.ts

```diff
@@ -109,6 +123,35 @@ export const createTenantsQueries = (client: Queryable<PostgreSql>) => {
     return rows;
   };
 
+  const deleteM2MApplicationByTenantId = async (tenantId: string) => {
+    await client.query(sql`
+      delete from applications where custom_client_metadata->>'tenantId' = ${tenantId} and tenant_id = ${adminTenantId} and "type" = ${ApplicationType.MachineToMachine}
+    `);
+  };
+
+  const deleteResourcesByTenantId = async (tenantId: string) => {
+    await client.query(sql`
+      delete from resources
+      where tenant_id = ${adminTenantId} and indicator = ${getManagementApiResourceIndicator(
+      tenantId
+    )}
+    `);
+  };
+
+  const deleteRoleByTenantId = async (tenantId: string) => {
+    await client.query(sql`
+      delete from roles
+      where tenant_id = ${adminTenantId} and name = ${getManagementApiAdminName(tenantId)}
+    `);
+  };
```

Comment:
This query method name is misleading, like the questions Charles asked above.  Could have a better name.  As all those resources belong to the admin Tenant.